src/types/FireOrBufferWithNames.ts

```diff
@@ -0,0 +1,4 @@
+export type FileOrBufferWithNames = {
+  file: Buffer | File;
+  name: string;
```

Comment:
Lets make this field optional.

With this change, when uploading the objects please make these changes:

- [ ] If the object is a `File` object && the `name` field is `undefined` use the `File.name` property
- [ ] If the object is a `File` object && `name` field is not `undefined` use the passed file name
- [ ] If the object is not a `File` object && the `name` field is undefined, throw an error (`FileNameMissingError`)